src/components/AgenciesPage/reducers/agenciesReducer.ts

```diff
@@ -0,0 +1,35 @@
+import { AgenciesActionTypes } from '../types/Agencies';
+import type { AgenciesState, AgenciesAction } from '../types/Agencies';
+
+const initialState: AgenciesState = {
+  agencies: [],
+  isLoading: false,
+  error: '',
+};
+
+export const agenciesReducer = (
+  state = initialState,
+  action: AgenciesAction,
+) => {
+  switch (action.type) {
+    case AgenciesActionTypes.FETCH_AGENCIES_REQUEST:
+      return {
+        ...state,
+        isLoading: true,
+      };
+    case AgenciesActionTypes.FETCH_AGENCIES_SUCCESS:
+      return {
+        agencies: action.payload,
+        isLoading: false,
+        error: '',
+      };
```

Comment:
1. copy ...state
2. you don't need to add `error: ""` on success action

Reply:
discussed, should be implemented in another story

Reply:
1.done
2. moved `error: ''` to request